src/devices/src/virtio/vsock/csm/connection.rs

```diff
@@ -655,6 +655,17 @@ where
     }
 }
 
+#[cfg(test)]
+impl<S> VsockConnection<S>
```

Comment:
Does it makes sense to add this impl inside the `mod tests` and then make the module `pub`?

Reply:
We could do that, but I think that this is the least invasive way of accessing private members declared in `connection.rs`

Reply:
I do not have a strong preference. My idea was to group all the tests relevant information in the `mod tests`, but I can not say that keeping this implementation outside the `mod tests` would counterfeit my idea of grouping. Eventually, we can group in the future such implementations for different structs in the `mod tests`, if they increase in number.

Reply:
> We could do that, but I think that this is the least invasive way of accessing private members declared in `connection.rs`

You can extend the struct implementation in `mod tests` without doing any other change and still be able to access private members; also the test module doesn't need to be `pub`. It's nice to keep the testing functionality in tests module because those functions won't be included in the `kcov` report, as opposed to how it is now implemented.

Reply:
Yep, you're right, that works too.

Reply:
As far as I understood, this impl is used outside the mod tests, so the private members are not available there. That's why the methods must be declared as pub(crate) (combined with the fact that the struct is public).

Otherwise, if the impl would be included inside the mod tests, the module would need to be public in order to make use from the public interface exposed by the implementation outside the module/file scope.